src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/ComponentParameterNullableWarningSuppressorTests.cs

```diff
@@ -0,0 +1,178 @@
+                DiagnosticResult.CompilerWarning("CS8618").WithSpan(10, 19, 10, 30).WithSpan(10, 19, 10, 30).WithArguments("property", "MyParameter").WithIsSuppressed(true)
+                );
+        }
+
+        [Fact]
+        public async Task LocallyDefinedAttributes()
+        {
+            var testCode = """
+
+                #nullable enable
+                using System;
+                using Microsoft.AspNetCore.Components;
+                
+                public class MyComponent
+                {
+                    [Parameter, EditorRequired]
+                    public string MyParameter { get; set; }
+                }
+
+                namespace Microsoft.AspNetCore.Components
+                {
+                    public class ParameterAttribute : Attribute { }
+                    public class EditorRequiredAttribute : Attribute { }
+                }
+
+                """;
+
+            await VerifyAnalyzerAsync(testCode, ReferenceAssemblies.Net.Net80,
+                // /0/Test0.cs(9,19): warning CS8618: Non-nullable property 'MyParameter' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+                DiagnosticResult.CompilerWarning("CS8618").WithSpan(9, 19, 9, 30).WithSpan(9, 19, 9, 30).WithArguments("property", "MyParameter").WithIsSuppressed(true)
+                );
+        }
+
+        [Fact]
+        public async Task LocallyDefinedAttributesDifferentNamespace()
+        {
+            var testCode = """
+
+                #nullable enable
+                using System;
+                using MyNamespace;
+                
+                public class MyComponent
+                {
+                    [Parameter, EditorRequired]
+                    public string MyParameter { get; set; }
+                }
+
+                namespace MyNamespace
+                {
+                    public class ParameterAttribute : Attribute { }
+                    public class EditorRequiredAttribute : Attribute { }
+                }
+
+                """;
+
+            await VerifyAnalyzerAsync(testCode, ReferenceAssemblies.Net.Net80,
+                // /0/Test0.cs(9,19): warning CS8618: Non-nullable property 'MyParameter' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+                DiagnosticResult.CompilerWarning("CS8618").WithSpan(9, 19, 9, 30).WithSpan(9, 19, 9, 30).WithArguments("property", "MyParameter")
+                );
+        }
+
+        private static Task VerifyAnalyzerAsync(string source, params DiagnosticResult[] expected)
+            => VerifyAnalyzerAsync(source,
+                                   ReferenceAssemblies.Net.Net80.WithAssemblies([typeof(EditorRequiredAttribute).Assembly.Location[0..^4]]),
```

Comment:
Can't we use AspNet reference assemblies rather than this?

Reply:
There isn't an AspNet set of reference assemblies. I think jared's version of them has it though, let me pull that in and try it.

Reply:
I see them in the package version list:

https://github.com/dotnet/razor/blob/c0bd75d99369adcd5a2f7e1f1ac42bee8a3bf619/Directory.Packages.props#L24

Reply:
Ugh, they don't work with the testing framework. The framework actually loads stuff of off disk, as opposed to being in memory, and there doesn't seem to be anyway to pass it by memory. I suppose we could write out the basic ref assembly to disk to at least make it deterministic.